### ISSUE TITLE
Changing printer profile resets all filament, colors and process to last remembered ones for selected printer profile

### ISSUE THREAD

**Issue by r0mar0ma** [SoftFever/OrcaSlicer#5332]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.0.0

### Operating System (OS)

Windows

### OS Version

Windows 11

### Printer

Bambulab X1C

### How to reproduce

1. Open any .3mf project with multiple filaments and color paintings and printer X1C
2. Switch to another printer profile P1S

![step0](https://github.com/SoftFever/OrcaSlicer/assets/76665418/0a1b34b0-4006-4c6d-8d2c-ed4383a9381d)
![step1](https://github.com/SoftFever/OrcaSlicer/assets/76665418/9da63d55-98cf-4ff1-b5d8-bcbaa2a1b44b)
![step2](https://github.com/SoftFever/OrcaSlicer/assets/76665418/ce46e66b-745b-442f-967b-379e14542739)
![step3](https://github.com/SoftFever/OrcaSlicer/assets/76665418/6e5f35d1-4a08-4c65-8999-d1c05abbb3b8)

### Actual results

After switched to another printer profile:
- ALL selected filament profiles resets to last ones that was used before with new selected P1S profile
- ALL coloring removed
- process switched to another one

### Expected results

Should not associate, remember and reset filament profiles and process profiles when changing printer profile, like it's working in BambuStudio.
Should not reset coloring.

### Project file & Debug log uploads

[Cylinder.zip](https://github.com/SoftFever/OrcaSlicer/files/15282380/Cylinder.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

### Anything else?

There is no any way to print multimaterial files with coloring because changing of printer profile from the one which is saved into the project to the one which I want to use just resets all filaments and colorings.

BambuStudio is not remembering "last used" filaments and process for the printer which is much more correct way.

If this remembering is "the feature" then probably should be the an option to disable this. Current workflow with remembering is making OrcaSlicer unusable to print such projects.

**Comment by pennystockss** [comment]
Yes absolutely yes need this feature. Changing between printers is super frustrating when we have to change all the colors, let alone if new printer doesn't have enough colors need to add them, then reload 3mf file. If you forget to do that then save you erased your painting.

**Comment by rdbischoff** [comment]
pointless to share 3MFs without being able to switch to your printer without keeping the color painting and such.

**Comment by witchrobin** [comment]
I also ran into the same problem when trying to open a multicolor print from Makerworld where it would reset all the filament from the 3mf file once I change printer.  Found out there is an option in preferences to "Remember printer configuration".  Turning that off fixed the problem and behaves like BS.

**Comment by L2on1** [comment]
dub of #4145 it explains well the same issue with painting for instance and #5446 , #3898

**Comment by kemper** [comment]
I've mostly been using Orca since getting a Bambu P1S and thought that model creators were just doing something dumb with their 3MF files. Then I decided to google a bit and realized this happens with Orca and not Bambu Studio. Glad I have a workaround but this means I'm probably not going to use Orca since I download so many 3mf files.

**Comment by slooker** [comment]
I'm having the same problem, unfortunately.  If I have to change the printer choice, it drops all colors and assigns random colors.  I have "Remember printer configuration" checked.  
![image](https://github.com/user-attachments/assets/44518d48-eb3e-4dda-b242-3f1961ebf1d2)
![image](https://github.com/user-attachments/assets/3bcde27f-ce21-461e-9c7e-41abca4da45e)
![image](https://github.com/user-attachments/assets/3faaceb8-c012-4664-9dd1-3cd97a77552c)

**Comment by CaptianNoob** [comment]
> I also ran into the same problem when trying to open a multicolor print from Makerworld where it would reset all the filament from the 3mf file once I change printer. Found out there is an option in preferences to "Remember printer configuration". Turning that off fixed the problem and behaves like BS.

This is exactly what I needed. Thank you good sir.

**Comment by Lomax21** [comment]
> I also ran into the same problem when trying to open a multicolor print from Makerworld where it would reset all the filament from the 3mf file once I change printer. Found out there is an option in preferences to "Remember printer configuration". Turning that off fixed the problem and behaves like BS.

This 100% solved the issue... I was having to open files in BS to switch from A1 to my P and X's just to save the profile settings much less the filament colors. After unchecking that option, it works like it should now.

**Comment by vinny1575** [comment]
> I also ran into the same problem when trying to open a multicolor print from Makerworld where it would reset all the filament from the 3mf file once I change printer. Found out there is an option in preferences to "Remember printer configuration". Turning that off fixed the problem and behaves like BS.

Thank you for this!

**Comment by adfaklsdjf** [comment]
does that mean this is the intended behavior?  it's hard to imagine why i'd want to use "Remember printer configuration" setting.  

@witchrobin you're a lifesaver.

**Comment by soundfx37** [comment]
> I also ran into the same problem when trying to open a multicolor print from Makerworld where it would reset all the filament from the 3mf file once I change printer. Found out there is an option in preferences to "Remember printer configuration". Turning that off fixed the problem and behaves like BS.

This is the correct answer!

**Comment by adfaklsdjf** [comment]
Still seems like a bug to me... ¯\_(ツ)_/¯

**Comment by FelixBuehler** [comment]
Same, at least with the latest nightly build, I can't change the printer without losing all filament information. The checkbox "Remember printer configuration" is set.

**Comment by kemper** [comment]
I used gemini to research and created a proposed fix. I haven't been able to successfully build the project to test it, but the change seems reasonable.

**Comment by M2ABRAMSTANK** [comment]
Same issue here, disabling remember printer configuration avoids the reset, works correctly in Bambu Studio either way. Same on both latest nightly & stable
